src/Raven.Server/Documents/ETL/Providers/S3/S3Etl.cs

```diff
@@ -0,0 +1,171 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Parquet;
+using Parquet.Data;
+using Raven.Client.Documents.Changes;
+using Raven.Client.Documents.Operations.Counters;
+using Raven.Client.Documents.Operations.ETL;
+using Raven.Server.Documents.ETL.Providers.SQL;
+using Raven.Server.Documents.Replication.ReplicationItems;
+using Raven.Server.Documents.TimeSeries;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Documents.ETL.Providers.S3
+{
+    public class S3Etl : EtlProcess<ToS3Item, RowGroups, S3EtlConfiguration, S3ConnectionString>
+    {
+        public const string S3EtlTag = "S3 ETL";
+
+        public readonly S3EtlMetricsCountersManager S3Metrics = new S3EtlMetricsCountersManager();
+
+        private Timer _timer;
+
+        public S3Etl(Transformation transformation, S3EtlConfiguration configuration, DocumentDatabase database, ServerStore serverStore)
+            : base(transformation, configuration, database, serverStore, S3EtlTag)
+        {
+            Metrics = S3Metrics;
+
+            _timer = new Timer(_ => _waitForChanges.Set(), null, TimeSpan.Zero, configuration.ETLFrequency);
```

Comment:
If I understand this properly, we are handling the delay of this kind of ETL using the timer?
So we'll start running on startup, then every `ETLFrequency` ?

Reply:
This is a good point. 
We need to persist last time it was run and adjust the timer, so we will not start on each server restart.

Reply:
Take into account that this may _move_ between nodes as well, so we have to account for that as well.